EventsExpress.Core/Services/EventService.cs

```diff
@@ -58,12 +58,58 @@ public async Task<OperationResult> AddUserToEvent(Guid userId, Guid eventId)
                 ev.Visitors = new List<UserEvent>();
             }
 
-            ev.Visitors.Add(new UserEvent { EventId = eventId, UserId = userId });
+            if (ev.IsPublic)
+            {
+                ev.Visitors.Add(new UserEvent { EventId = eventId, UserId = userId, UserStatusEvent = 0 });
+            }
+            else
+            {
+                ev.Visitors.Add(new UserEvent { EventId = eventId, UserId = userId, UserStatusEvent = (UserStatusEvent)2 });
```

Comment:
Consider using enum values here, rather than its numeric representation. The code is unreadable in this way.